src/mon/MonClient.cc

```diff
@@ -625,17 +625,21 @@ MonConnection& MonClient::_add_conn(unsigned rank)
 
 void MonClient::_add_conns()
 {
-  unsigned n = cct->_conf->mon_client_hunt_parallel;
-  if (n == 0 || n > monmap.size()) {
-    n = monmap.size();
-  }
-  vector<unsigned> ranks(n);
-  for (unsigned i = 0; i < n; i++) {
+  unsigned m = monmap.size();
+  vector<unsigned> ranks(m);
+  for (unsigned i = 0; i < m; i++) {
     ranks[i] = i;
   }
   std::random_device rd;
   std::mt19937 rng(rd());
   std::shuffle(ranks.begin(), ranks.end(), rng);
+
+  unsigned n = cct->_conf->mon_client_hunt_parallel;
+  if (n) {
```

Comment:
could you keep the original block of `if (n == 0 || n > monmap.size())`, IMHO, it's more explicit and easier to digest